plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -1904,7 +1924,22 @@ protected StartAnswer execute(StartCommand cmd) {
                 }
             }
 
+            // The number of disks changed must be 0 for install as is, as the VM is a clone
+            int disksChanges = !installAsIs ? disks.length : 0;
             int totalChangeDevices = disks.length + nics.length;
+            int hackDeviceCount = 0;
+            if (diskInfoBuilder != null) {
+                hackDeviceCount += diskInfoBuilder.getDiskCount();
+            }
+            hackDeviceCount += nicDevices == null ? 0 : nicDevices.length;
+            // vApp cdrom device
+            // HACK ALERT: ovf properties might not be the only or defining feature of vApps; needs checking
```

Comment:
@nvazquez this is not necessary if we're cloning a deploy as-is template, there is no need to add the hack to add another cdrom device. Please check and remove this code.